photon-ml/src/main/scala/com/linkedin/photon/ml/Driver.scala

```diff
@@ -169,19 +169,26 @@ protected[ml] class Driver(
   }
 
   protected def prepareTrainingData(): Unit = {
+    // Verify that selected features files exist
     params.selectedFeaturesFile.foreach(file => {
       val path = new Path(file)
       if (! path.getFileSystem(sc.hadoopConfiguration).exists(path)) {
         throw new IOException("Could not find [" + file + "]. Check that the file exists")
       }
     })
+
+    // Read training data
     trainingData = suite.readLabeledPointsFromAvro(
       sc, params.trainDir, params.selectedFeaturesFile, params.minNumPartitions)
       .persist(trainDataStorageLevel).setName("training data")
-    featureNum = trainingData.first().features.size
+
     trainingDataNum = trainingData.count().toInt
+    require(trainingDataNum > 0,
+      "No training data found. Ensure that training data exists and feature vectors are not empty.")
```

Comment:
Thanks for making the error reporting more informative.

One suggestion:  ```trainingDataNum > 0``` and "feature vectors are not empty" are two independent things, because we can still have the case where ```trainingDataNum > 0``` while ```featureNum == 0```. Shall we decouple those two things from the error message, and only say "No training data found. Ensure that training data exists."?